### PR TITLE
Setting configuration service key

### LINKED ISSUE #105: `SEGFAULT/SIGSEGV` when deploying core contracts by default via Docker
### Problem

While using Docker Compose to deploy the emulator it seg faults (`SIGSEGV`) and aborts when trying to deploy core contracts by default via environment variables.

### Steps to Reproduce

Use the following `docker-compose.yml` (truncated to exclude other services that aren't relevant):

```
version: '3'
services:
  flow-dev-wallet:
    image: ghcr.io/onflow/fcl-dev-wallet:latest
    ports:
      - '8701:8701'
    environment:
      - PORT=8701
      - FLOW_ACCESS_NODE=http://flow-blockchain-emulator:8081
      - FLOW_ACCOUNT_KEY_ID=0
      - FLOW_ACCOUNT_PRIVATE_KEY=e22f344c8d8218effc51025bb690bddd52721ed6bbf815b951fcbd48de2c6a5c
      - FLOW_ACCOUNT_PUBLIC_KEY=01910f53695c0b5220076f6496d0a1dcf7a4c202768efcffe86c86a04eea60f82c61cbb029f6651282c75086bf7d2246cec6795244e6518652f375fea808a407
      - FLOW_INIT_ACCOUNTS=0
      - FLOW_ACCOUNT_ADDRESS=0xf8d6e0586b0a20c7
      - FLOW_AVATAR_URL=https://avatars.onflow.org/avatar/
  flow-blockchain-emulator:
    image: gcr.io/flow-container-registry/emulator:0.27.2
    ports:
      - '8081:8081'
      - '3569:3569'
    environment:
      - FLOW_HTTPPORT=8081
      - FLOW_SERVICEPRIVATEKEY=e22f344c8d8218effc51025bb690bddd52721ed6bbf815b951fcbd48de2c6a5c
      - FLOW_SERVICEKEYSIGALGO=ECDSA_P256
      - FLOW_SERVICEKEYHASHALGO=SHA3_256
      - FLOW_WITHCONTRACTS=true
```

Note the use of the `FLOW_WITHCONTRACTS` in the environment variables for the `flow-blockchain-emulator` service. `flow-blockchain-emulator` is also pinned to version `0.27.2` which is presently the latest version of the Flow emulator that is published.

Run the following command:

```
docker-compose up -d
```

View logs of the `flow-blockchain-emulator` service and view seg fault:

```
docker-compose logs flow-blockchain-emulator

Attaching to super-octo-spork_flow-blockchain-emulator_1
flow-blockchain-emulator_1  | time="2021-12-13T17:22:22Z" level=info msg="⚙️   Using service account 0xf8d6e0586b0a20c7" serviceAddress=f8d6e0586b0a20c7 serviceHashAlgo=SHA3_256 servicePrivKey=e22f344c8d8218effc51025bb690bddd52721ed6bbf815b951fcbd48de2c6a5c servicePubKey=01910f53695c0b5220076f6496d0a1dcf7a4c202768efcffe86c86a04eea60f82c61cbb029f6651282c75086bf7d2246cec6795244e6518652f375fea808a407 serviceSigAlgo=ECDSA_P256
flow-blockchain-emulator_1  | time="2021-12-13T17:22:22Z" level=info msg="📜  Flow contract" FlowServiceAccount=0xf8d6e0586b0a20c7
flow-blockchain-emulator_1  | time="2021-12-13T17:22:22Z" level=info msg="📜  Flow contract" FlowToken=0x0ae53cb6e3f42a79
flow-blockchain-emulator_1  | time="2021-12-13T17:22:22Z" level=info msg="📜  Flow contract" FungibleToken=0xee82856bf20e2aa6
flow-blockchain-emulator_1  | time="2021-12-13T17:22:22Z" level=info msg="📜  Flow contract" FlowFees=0xe5a8b7f23e8b548f
flow-blockchain-emulator_1  | time="2021-12-13T17:22:22Z" level=info msg="📜  Flow contract" FlowStorageFees=0xf8d6e0586b0a20c7
flow-blockchain-emulator_1  | panic: runtime error: invalid memory address or nil pointer dereference
flow-blockchain-emulator_1  | [signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x7b092b]
flow-blockchain-emulator_1  | 
flow-blockchain-emulator_1  | goroutine 1 [running]:
flow-blockchain-emulator_1  | github.com/onflow/flow-go-sdk/crypto.InMemorySigner.Sign(...)
flow-blockchain-emulator_1  |   /go/pkg/mod/github.com/onflow/flow-go-sdk@v0.24.0/crypto/crypto.go:129
flow-blockchain-emulator_1  | github.com/onflow/flow-go-sdk.(*Transaction).SignEnvelope(0xc000642180, 0xc7200a6b58e0d6f8, 0x0, 0x188d7c0, 0xc001399920, 0x0, 0x0)
flow-blockchain-emulator_1  |   /go/pkg/mod/github.com/onflow/flow-go-sdk@v0.24.0/transaction.go:316 +0xb4
flow-blockchain-emulator_1  | github.com/onflow/flow-emulator/server.deployContract(0xc000641a40, 0x1507542, 0x4, 0xc0012e0000, 0x1cdf, 0x2000, 0x18b4c28, 0xc0001e6530)
flow-blockchain-emulator_1  |   /app/server/contracts.go:91 +0x37e
flow-blockchain-emulator_1  | github.com/onflow/flow-emulator/server.deployContracts(0xc000641a40, 0xc000000004, 0x1517d2c, 0x13, 0x0, 0x0)
flow-blockchain-emulator_1  |   /app/server/contracts.go:50 +0x528
flow-blockchain-emulator_1  | github.com/onflow/flow-emulator/server.NewEmulatorServer(0xc0001eb490, 0xc0000fe1e0, 0x150fde7)
flow-blockchain-emulator_1  |   /app/server/server.go:143 +0x945
flow-blockchain-emulator_1  | github.com/onflow/flow-emulator/cmd/emulator/start.Cmd.func1(0xc0002b0500, 0xc00013a740, 0x2, 0x2)
flow-blockchain-emulator_1  |   /app/cmd/emulator/start/start.go:177 +0x85e
flow-blockchain-emulator_1  | github.com/spf13/cobra.(*Command).execute(0xc0002b0500, 0xc00018c010, 0x2, 0x2, 0xc0002b0500, 0xc00018c010)
flow-blockchain-emulator_1  |   /go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:860 +0x2c2
flow-blockchain-emulator_1  | github.com/spf13/cobra.(*Command).ExecuteC(0xc0002b0500, 0xc0002b0500, 0x1297680, 0xc0000420b8)
flow-blockchain-emulator_1  |   /go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0x375
flow-blockchain-emulator_1  | github.com/spf13/cobra.(*Command).Execute(...)
flow-blockchain-emulator_1  |   /go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902
flow-blockchain-emulator_1  | main.main()
flow-blockchain-emulator_1  |   /app/cmd/emulator/main.go:46 +0x3b
```

### Acceptance Criteria

No segmentation fault and that the following core contracts are deployed:
* `FUSD`
* `NonFungibleToken`
* `NFTStorefront`

This is per this source: https://github.com/onflow/flow-emulator/blob/3192ba29f5489351126ddb09f41c5853319fe17f/server/contracts.go#L26

### Context

Blocks verification of deploying our contracts that depend on the aforementioned core contracts with a Flow emulated environment. This makes local development more difficult/cumbersome.

### PR DESCRIPTION
**Pull request by sideninja**
Closes #105 

## Description
Setting configuration service key was only setting provided public key but not private key which failed for deployment of contracts.

Dependencies were also updated and some minor warning resolved.
______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: blockchain.go [hunk 1]
```diff
@@ -452,8 +468,7 @@ func bootstrapLedger(
 
 	bootstrap := configureBootstrapProcedure(conf, flowAccountKey, conf.GenesisTokenSupply)
 
-	programs := programs.NewEmptyPrograms()
-	err := vm.Run(ctx, bootstrap, ledger, programs)
```
Comment (sideninja) [inline blockchain.go:456]: this is resolving import conflict warning

File: blockchain.go [hunk 2]
```diff
@@ -871,9 +886,7 @@ func (b *Blockchain) executeNextTransaction(ctx fvm.Context) (*types.Transaction
 		) (*fvm.TransactionProcedure, error) {
 			tx := fvm.Transaction(txBody, txIndex)
 
-			programs := programs.NewEmptyPrograms()
-
-			err := b.vm.Run(ctx, tx, ledgerView, programs)
```
Comment (sideninja) [inline blockchain.go:876]: resolving warning

File: blockchain.go [hunk 3]
```diff
@@ -932,11 +945,11 @@ func (b *Blockchain) commitBlock() (*flowgo.Block, error) {
 	if err != nil {
 		return nil, err
 	}
-	delta := b.pendingBlock.LedgerDelta()
```
Comment (sideninja) [inline blockchain.go:935]: resolving warning

File: blockchain.go [hunk 4]
```diff
@@ -1024,8 +1037,7 @@ func (b *Blockchain) ExecuteScriptAtBlock(script []byte, arguments [][]byte, blo
 
 	scriptProc := fvm.Script(script).WithArguments(arguments...)
 
-	programs := programs.NewEmptyPrograms()
-	err = b.vm.Run(blockContext, scriptProc, requestedLedgerView, programs)
```
Comment (sideninja) [inline blockchain.go:1028]: resolving warning

File: cmd/emulator/start/start.go [hunk 5]
```diff
@@ -29,7 +30,6 @@ import (
 	sdk "github.com/onflow/flow-go-sdk"
 	"github.com/onflow/flow-go-sdk/crypto"
 	"github.com/onflow/flow-go/fvm"
-	"github.com/prometheus/common/log"
```
Comment (sideninja) [inline cmd/emulator/start/start.go:32]: this was deprecated and no longer exists

**Review by janezpodhostnik — APPROVED** [review]
Nice cleanup.

**Pull request merged**